src/app/components/Input/InputRange.tsx

```diff
@@ -22,7 +22,12 @@ export const InputRange = React.forwardRef<HTMLInputElement, Props>(
 		const handleInput = (value: string) => {
 			const fraction = Math.pow(10, magnitude! * -1);
 			const amount = BigNumber.make(value).times(fraction);
-			setValues([amount.toNumber()]);
+
+			if (amount.isGreaterThan(max)) {
+				setValues([max]);
+			} else {
+				setValues([amount.toNumber()]);
```

Comment:
Hi, @brenopolanski !

Nice catch 🌮  I just think that we don't need this else block over here, you could simplify this complexity by returning the setValues with max inside the if and return the set values with the amount if it isn't greater than like:

```
if (amount.isGreaterThan(max)) return setValues([max]);

return setValues([amount.toNumber()];
```

Reply:
@clucasalcantara Don't you think it's a little strange to return 'setValues`? I mean, because `setValues` updates only the state `values`. What do you think @luciorubeens ?

Reply:
Why strange? The return is only to jump out the function when the branch decided to be executed is found (is greater than or not) but I can be wrong ¯\_(ツ)_/¯. I just think that we should add cyclomatic complexity in functions when we can use a direct execution, which is the same approach when you use a return inside a `switch` statement to avoid the usage of `break`.

Reply:
ahh @clucasalcantara  i understand now what do you mean :+1: